ros_ign_bridge/src/factory_interface.hpp

```diff
@@ -15,21 +15,24 @@
 #ifndef  FACTORY_INTERFACE_HPP_
 #define  FACTORY_INTERFACE_HPP_
 
-// include ROS 2
-#include <rclcpp/rclcpp.hpp>
-
 // include Ignition Transport
 #include <ignition/transport/Node.hh>
 
 #include <memory>
 #include <string>
 
+// include ROS 2
+#include <rclcpp/rclcpp.hpp>
+
 namespace ros_ign_bridge
 {
 
 class FactoryInterface
 {
 public:
+  virtual
+  ~FactoryInterface() = 0;
```

Comment:
```suggestion
  ~FactoryInterface();
```

It's not really pure because we're providing the implementation, right?

Reply:
You still have to provide a body for a pure-virtual destructor.  There is probably a more official resource, but this confirms: https://www.geeksforgeeks.org/pure-virtual-destructor-c/

Reply:
Oh gosh, ok. Interestingly, before posting the suggestion I tried it locally and it compiles fine (at least with gcc) :woman_shrugging:

Reply:
The only reason I caught it was because of templight, which uses clang under the hood.